src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java

```diff
@@ -0,0 +1,151 @@
+    public String wrapSuperTypeIntoList(MappingContext mappingContext, String type) {
+        return getGenericsString(mappingContext, KOTLIN_UTIL_LIST, "out " + type);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String wrapApiReturnTypeIfRequired(MappingContext mappingContext,
+                                              NamedDefinition namedDefinition,
+                                              String parentTypeName) {
+        String computedTypeName = namedDefinition.getJavaName();
+        if (parentTypeName.equalsIgnoreCase(GraphQLOperation.SUBSCRIPTION.name()) &&
+                Utils.isNotBlank(mappingContext.getSubscriptionReturnType())) {
+            // in case it is subscription and subscriptionReturnType is set
+            return getGenericsString(mappingContext, mappingContext.getSubscriptionReturnType(), computedTypeName);
+        }
+        // Consider only Java-kotlin OR Java-Scala cross language calls
+        if (Boolean.TRUE.equals(mappingContext.getUseOptionalForNullableReturnTypes()) && !namedDefinition.isMandatory()) {
+            if (!computedTypeName.startsWith(KOTLIN_UTIL_LIST) && !computedTypeName.startsWith(JAVA_UTIL_LIST)) {
+                // append `?` (except java list and kotlin list)
+                computedTypeName = getOptionString(mappingContext, computedTypeName);
+            }
+        }
+
+        if (computedTypeName.startsWith(KOTLIN_UTIL_LIST) &&
+                Utils.isNotBlank(mappingContext.getApiReturnListType())) {
+            // in case it is query/mutation, return type is list and apiReturnListType is set
+            return computedTypeName.replace(KOTLIN_UTIL_LIST, mappingContext.getApiReturnListType());
+        }
+        if (Utils.isNotBlank(mappingContext.getApiReturnType())) {
+            // in case it is query/mutation and apiReturnType is set
+            return getGenericsString(mappingContext, mappingContext.getApiReturnType(), computedTypeName);
+        }
+        return getTypeConsideringPrimitive(mappingContext, namedDefinition, computedTypeName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isPrimitive(String scalaType) {
+        return isKotlinPrimitive(scalaType);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getGenericsString(MappingContext mappingContext, String genericType, String typeParameter) {
+        return String.format("%s<%s>", genericType, typeParameter);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String mapDirectiveArgumentValue(MappingContext mappingContext, Argument dirArg, String argumentValueFormatter) {
+        return valueMapper.map(mappingContext, dirArg.getValue(), null, argumentValueFormatter);
+    }
+
+    @Override
+    public boolean addModelValidationAnnotationForType(String possiblyPrimitiveType) {
+        return !KotlinGraphQLTypeMapper.isKotlinPrimitive(possiblyPrimitiveType);
```

Comment:
Kotlin doesn't have primitive types. I'm not sure this is valid.
I don't remember if this library supports other validations than NotNull.
If we do support arbitrary validation, it should be done on any type (including 'primitive').
If we only support nullability, it is irrelevant in Kotlin because we have the type system for that already.

Reply:
Just to not verify null.if it's a `a:Int`, then, donot need generate if (a!=null)
Of course, the question mark is also judged. It can be deleted here, but in order to allow the annotation to be verified, it is still retained.

Reply:
I have trouble understanding what you're saying here. But yeah as far as nullability is concerned, there is no difference between primitive types and non-primitive types in Kotlin. Only `?` matters.

What annotation are you talking about? `javax.validation.NotNull`?
This one is also not needed in the Kotlin codegen.

Reply:
There is no disable nonull annotation, it is only turned off by default. Of course, if it is completely prohibited, it can be removed